### PR TITLE
Fix emailVerified bug with ConvertibleBoolean

### PR DESCRIPTION
**Pull request by ianthpun**
Create a new ConvertibleBoolean type that handles both string and bool type marshal/unmarshalling.

Comes with a bucket of tests to prove it works

### CODE REVIEW

File: custom.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package auth0
```
Comment (ianthpun) [inline custom.go]: not married to having this type inside `package auth0`, opening to suggestions where to move it (`internal/custom`)?

Comment (ianthpun) [reply custom.go]: moved this back to `management/users`

File: management/user.go [hunk 2]
```diff
@@ -87,6 +87,50 @@ type User struct {
...
+	return nil
+}
+
+// MarshalJSON handles marshalling of the ConvertibleBoolean value. We will always return the value without quotes.
+func (bit ConvertibleBoolean) MarshalJSON() ([]byte, error) {
+	return []byte(strconv.FormatBool(bool(bit))), nil
+}
+
+func trimQuotes(s string) string {
+	if len(s) > 0 && s[0] == '"' {
```
Comment (bfbachmann) [inline management/user.go:125]: Minor: could also do
```go
	strings.Replace(s, `"`, "")
```

**Review by bfbachmann — APPROVED** [review]
Nice work~

**Pull request merged**